### ISSUE TITLE
[Rule change] ignore method invocations in a variable declaration for `prefer-moving-to-variable`

### ISSUE THREAD

**Issue by w3ggy** [dart-code-checker/dart-code-metrics#1142]
First of all. Thanks for amazing tool! We use [flutter_hooks](https://pub.dev/packages/flutter_hooks) package on our project. When we add the `prefer-moving-to-variable` rule we see a lot of false-positive warnings related to hooks.

<details>
<img width="738" alt="image" src="https://user-images.githubusercontent.com/12301508/211306073-faebe9cd-bcb4-41c1-8854-ff3e2cd7a22f.png">
</details>

**What rule do you want to change?**
prefer-moving-to-variable

**How will the change be implemented? (New option, new default behavior, etc.)?**
Honestly, I don't know. It looks like we should use naming for omitting some function calls.

**Please provide some example code that this change will affect:**

```dart
    final key1 = useMemoized(() => GlobalKey<InputFieldState>(), []);
    final key2 = useMemoized(() => GlobalKey<InputFieldState>(), []);
    final key3 = useMemoized(() => GlobalKey<InputFieldState>(), []);
    final key4 = useMemoized(() => GlobalKey<InputFieldState>(), []);
    final key5 = useMemoized(() => GlobalKey<InputFieldState>(), []);
```

**What does the rule currently do for this code?**

Warns `Prefer moving repeated invocations to variable and use it instead. (Documentation)`

**What will the rule do after it's changed?**

It should skip hook validations

**Are you willing to submit a pull request to implement this change?**

I don't know how.

**Comment by incendial** [comment]
@w3ggy can you provide a repro? To understand the context in which keys are used.

**Comment by w3ggy** [comment]
@incendial complete pseudo-sample of the form. I removed all unnecessary logic to make it simple. Let me know if you need something else.

```dart
class CreateNewAccountPage extends HookWidget {
  const CreateNewAccountPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final focusNodeEmail = useFocusNode();
    final focusNodeUsername = useFocusNode();
    final focusNodeFirstName = useFocusNode();
    final focusNodeLastName = useFocusNode();
    final focusNodeDayOfBirth = useFocusNode();
    final focusNodeMobileNumber = useFocusNode();

    final textControllerEmail = useTextEditingController();
    final textControllerUserName = useTextEditingController();
    final textControllerFirstName = useTextEditingController();
    final textControllerLastName = useTextEditingController();
    final textControllerDayOfBirth = useTextEditingController();
    final textControllerMobileNumber = useTextEditingController();

    return Scaffold(
      backgroundColor: Colors.white,
      body: ListView(
        padding: const EdgeInsets.symmetric(horizontal: 24),
        children: [
          TextFormField(
            controller: textControllerEmail,
            focusNode: focusNodeEmail,
          ),
          TextFormField(
            controller: textControllerUserName,
            focusNode: focusNodeUsername,
          ),
          TextFormField(
            controller: textControllerFirstName,
            focusNode: focusNodeFirstName,
          ),
          TextFormField(
            controller: textControllerLastName,
            focusNode: focusNodeLastName,
          ),
          TextFormField(
            controller: textControllerDayOfBirth,
            focusNode: focusNodeDayOfBirth,
          ),
          TextFormField(
            controller: textControllerMobileNumber,
            focusNode: focusNodeMobileNumber,
          ),
        ],
      ),
    );
  }
}
```

And it seems strange that the linter highlights only the `useFocusNode/useMemoized` but not the `useTextEditingController`

<details>
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/12301508/211328356-c28277a0-e741-4e72-b10c-3ea84d9ffb00.png">
</details>

I guess it is because the `useFocusNode` is just function invokation under the hood 

<details>
<img width="458" alt="image" src="https://user-images.githubusercontent.com/12301508/211328535-7d777948-c661-44b6-9a63-3017ea39b9cc.png">
</details> and `useTextEditingController` is constructor invocation <details>
<img width="552" alt="image" src="https://user-images.githubusercontent.com/12301508/211328680-f049db60-de67-4c42-84f0-a794557eae1f.png">
</details>

**Comment by incendial** [comment]
> and `useTextEditingController` is constructor invocation

Yes, you're right.

So, looks like the rule should ignore an invocation if it's already a part of the assignment.

**Comment by incendial** [comment]
And thank you for the details

**Comment by incendial** [comment]
Fixed in 5.5.0 🚀
